test/features/edit_user_test.rb

```diff
@@ -0,0 +1,19 @@
+require "test_helper"
+
+class EditUserTest < Capybara::Rails::TestCase
+
+  def user
+    @user ||= users :normal
+    @user.password = 'password'
+    @user
+  end
+
+  test "signed in user can edit their profile by changing their name" do
+    sign_in user, visit: true, submit: true
+    visit edit_user_registration_path
+    fill_in 'user_first_name', :with => 'William'
+    fill_in 'user_current_password', :with => user.password
```

Comment:
Sure https://github.com/plataformatec/devise/wiki/How-To%3a-Allow-users-to-edit-their-account-without-providing-a-password